Sources/Pool.swift

```diff
@@ -0,0 +1,7 @@
+public protocol Pool {
+    associatedtype Poolable
+    func borrow() -> Poolable?
+    func takeBack(poolable: Poolable)
```

Comment:
I think we have a disconnected logic between who's doing the action in these methods.

`borrow` sounds like the caller is borrowing something from the pool. By that logic `takeBack` should probably be `giveBack`.